pkg/cmd/optimize.go

```diff
@@ -124,8 +130,12 @@ var optimizeCmd = &cobra.Command{
 					continue
 				}
 
+				if len(values) < resultLimit || resultLimit == 0 {
+					resultLimit = len(values)
+				}
```

Comment:
select the slice only when the length exceeds the resultLimit?

if len(values) > resultLimit {
    values = values[:resultLimit]
}

Reply:
done